aggregation_mode/gateway/src/http.rs

```
@@ -322,4 +329,42 @@ impl GatewayServer {
                 .json(AppResponse::new_unsucessfull("Internal server error", 500)),
         }
     }
+
+    async fn get_quotas(req: HttpRequest) -> impl Responder {
+        let Some(state) = req.app_data::<Data<GatewayServer>>() else {
+            return HttpResponse::InternalServerError().json(AppResponse::new_unsucessfull(
+                "Internal server error: Failed to get app data",
+                500,
+            ));
+        };
+
+        let state = state.get_ref();
+
+        let Some(address_raw) = req.match_info().get("address") else {
+            return HttpResponse::BadRequest()
+                .json(AppResponse::new_unsucessfull("Missing address", 400));
+        };
+
+        // Check that the address is a valid ethereum address
+        if alloy::primitives::Address::from_str(address_raw.trim()).is_err() {
+            return HttpResponse::BadRequest()
+                .json(AppResponse::new_unsucessfull("Invalid address", 400));
+        }
+
+        let address = address_raw.to_lowercase();
+
+        let Ok(daily_tasks_by_address) = state.db.get_daily_tasks_by_address(&address).await else {
+            return HttpResponse::InternalServerError()
+                .json(AppResponse::new_unsucessfull("Internal server error", 500));
+        };
+
+        let formatted_time_left = get_time_left_day_formatted();
+
+        HttpResponse::Ok().json(AppResponse::new_sucessfull(serde_json::json!({
+            "proofs_submitted": daily_tasks_by_address,
+            "quota_limit": state.config.max_daily_proofs_per_user,
+            "quota_remaining": (state.config.max_daily_proofs_per_user - daily_tasks_by_address),
+            "quota_resets_in": formatted_time_left.as_str()
+        })))
```

Shouldn't we check that the user has paid? If not, the `quota_remaining` and `quota_limit` should be 0

---

I think that in that case, the user may detect that behavior as a failure on our side, so we should return an error indicating that the user is not subscribed.

---

I agrre, you can still send `200` but modify the message of the response

---

Done in [f8dea3a](https://github.com/yetanotherco/aligned_layer/pull/2202/commits/f8dea3a6ba0dcaceb709c0c28dd8786f82115516)